libs/vkd3d/meta.c

```diff
@@ -324,20 +513,327 @@ VkExtent3D vkd3d_get_clear_image_uav_workgroup_size(VkImageViewType view_type)
     }
 }
 
+HRESULT vkd3d_copy_image_ops_init(struct vkd3d_copy_image_ops *meta_copy_image_ops,
+        struct d3d12_device *device)
+{
+    VkDescriptorSetLayoutBinding set_binding;
+    VkPushConstantRange push_constant_range;
+    VkResult vr;
+    int rc;
+
+    memset(meta_copy_image_ops, 0, sizeof(*meta_copy_image_ops));
+
+    if ((rc = pthread_mutex_init(&meta_copy_image_ops->mutex, NULL)))
+    {
+        ERR("Failed to initialize mutex, error %d.\n", rc);
+        return hresult_from_errno(rc);
+    }
+
+    set_binding.binding = 0;
+    set_binding.descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_IMAGE;
```

Comment:
Shouldn't this be sampled image? The shaders use texture2D etc, not image2D.

Reply:
Yeah, typo.

Reply:
Fixed.